app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,73 @@
+class TweetController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?(session)
+      @user = User.find_by_id(session[:user_id])
+      erb :"/tweets/index"
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?(session)
+      @user = User.find_by_id(session[:user_id])
+      erb :'tweets/new'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets/new' do
+    if params[:content] != nil && params[:content] != ""
+      user = User.find_by(:id => session[:user_id])
+      tweet = Tweet.create(:content => params[:content], :user_id => user.id)
+      redirect "/tweets"
+    else
+      erb :"tweets/new"
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?(session)
+      @tweet = Tweet.find_by(:id => params[:id])
+      erb :"tweets/show"
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?(session)
+      if params[:id].to_i == session[:user_id]
```

Comment:
This is a little odd and i don't think it accomplishes what you intended it to.

Here, `params[:id]` is the tweet id. You are saying that a user can only edit a tweet, it if has the same ID as the user. We really want to be finding the tweet, and then checking if it belongs to the user before letting them edit:

```ruby
@tweet = Tweest.find_by(id: params[:id])
if  @tweet.user == current_user
```